profiles/utils_test.py

```diff
@@ -76,21 +78,40 @@ def test_too_long_prefix(user):
 @pytest.mark.django_db
 @pytest.mark.parametrize('url', [None, 'http://testserver/profiles/image.jpg'])
 @pytest.mark.parametrize('image', [None, Image.new('RGBA', size=(50, 50), color=(155, 0, 0))])
-def test_profile_img_url(url, image):
+@pytest.mark.parametrize('fullname', [None, 'Test User'])
+def test_profile_img_url(url, image, fullname):
     """
     Test that the correct profile image URL is returned.
     """
     profile = UserFactory.create().profile
     profile.image_small = url
+    profile.name = fullname
     if image:
         profile.image_small_file.save('/profiles/realimage.jpg', BytesIO(image.tobytes()), True)
     else:
         profile.image_small_file = None
     profile.save()
 
-    assert image_uri(profile.user, 'image_small') == (
-        profile.image_small_file.url if image else url if url else default_profile_image
+    profile_image = image_uri(profile.user, 'image_small')
+    assert profile_image.startswith(
+        profile.image_small_file.url if image else url if url else 'https://www.gravatar.com/avatar/'
     )
+    if not image and not url:
+        params_d = parse_qs(urlparse(profile_image).query)['d'][0]
+        if fullname:
+            assert params_d.endswith('profile/{}/64/fff/579cf9.png'.format(profile.user.username))
+        else:
+            assert params_d.endswith(DEFAULT_PROFILE_IMAGE)
```

Comment:
The forked logic in this test makes this pretty difficult to reason about (I know some of it existed before this PR). Can you split this into as many test cases as it takes to get rid of all the booleans? 1 case each for the following scenarios (maybe more): (1) profile has an image URL, (2) profile has an image, (3) profile has neither image URL nor image. (3) could be parametrized so that you pass in the user's username, profile name, and expected `d` querystring parameter value